### PR TITLE
Implement #657: Documentation: Examples for Immutable Collections.

### PR DESCRIPTION
**Pull request by saurabhmj**
Modified guide.md and added examples for the following cases:

    1. Creating Immutable Collections using .toImmutable()
    2. Modifying Immutable Collections with methods like newWith(), newWithAll(), ...

The general tone and examples are referred from the blog that @donraab had shared in the issue discussion. Thank you for the informative article!

Signed-off-by: saurabhmj <saurabhjogalekar@gmail.com>

### CODE REVIEW

File: docs/guide.md [hunk 1]
```diff
@@ -2593,6 +2641,18 @@ There are two ways to create an immutable **List**, **Set**, **Bag**, **Stack**
...
+    Lists.mutable.with("a", "b", "a", "c").toSet().toImmutable(); // creates a set of elements a,b,c
+ImmutableSet<String> set1 =  
+    Sets.mutable.with("a", "b", "a", "c").toImmutable(); // creates a set of elements a,b,c
+ImmutableMap<Integer, String> map = 
+    Maps.mutable.with(1, "a", 2, "b", 3, "c").toImmutable(); // creates a map with keys 1,2,3
+
+//2. Using factory classes
 ImmutableList<String> emptyList_i = 
     Lists.immutable.empty();  // creates an empty list
 ImmutableList<String> list_b =
```
Comment (nikhilnanivadekar) [inline docs/guide.md]: While you are updating the documentation, please add methods to create `ImmutableSet` and `ImmutableMap` using factory classes. New users might get confused if that is not referred. Thank you!

**Comment by nikhilnanivadekar** [comment]
Hi @saurabhmj can you please add the review comment, squash the commits to 1, rebase and push. I would like to merge it. Thank you again for your contribution!

**Comment by saurabhmj** [comment]
Hey sure @nikhilnanivadekar 

My apologies for the delay. Will try and do it this weekend.

**Comment by saurabhmj** [comment]
Hi @nikhilnanivadekar, I've pushed the changes. Let me know if anything else is missing. Would gladly add it. Thanks!

File: docs/guide.md [hunk 2]
```diff
@@ -2593,12 +2641,24 @@ There are two ways to create an immutable **List**, **Set**, **Bag**, **Stack**
...
+    Maps.mutable.with(1, "a", 2, "b", 3, "c").toImmutable(); // creates a map with keys 1,2,3
+
+//2. Using factory classes
 ImmutableList<String> emptyList_i = 
     Lists.immutable.empty();  // creates an empty list
-ImmutableList<String> list_b = 
-    Lists.immutable.with("a", "b", "c"); // creates a list of elements a,b,c
-ImmutableList<String> list_c = 
-    Lists.immutable.of("a", "b", "c");  // creates a list of elements a,b,c
+ImmutableSet<String> set =
```
Comment (nikhilnanivadekar) [inline docs/guide.md]: That's not quite what I was expecting. Let me write it down what I was expecting. All these 3 should be present for clarity.

```
ImmutableList<String> emptyList_i = 
    Lists.immutable.empty();  // creates an empty list
ImmutableList<String> list_b = 
    Lists.immutable.with("a", "b", "c"); // creates a list of elements a,b,c
ImmutableList<String> list_c = 
    Lists.immutable.of("a", "b", "c");  // creates a list of elements a,b,c

ImmutableSet<String> emptySet_i = 
    Sets.immutable.empty();  // creates an empty Set
ImmutableSet<String> Set_b = 
    Sets.immutable.with("a", "b", "c"); // creates a Set of elements a,b,c
ImmutableSet<String> Set_c = 
    Sets.immutable.of("a", "b", "c");  // creates a Set of elements a,b,c

ImmutableMap<Integer, String> map = 
    Maps.immutable.empty(); // creates an empty map
ImmutableMap<Integer, String> map = 
    Maps.immutable.with(1, "a", 2, "b", 3, "c"); // creates a map with keys 1,2,3

```

**Comment by nikhilnanivadekar** [comment]
@saurabhmj apologies for the miscommunication, I have mentioned what I think we should add to the guide to make it clearer to the new users.

**Comment by saurabhmj** [comment]
Ah, my bad.
Thanks for the suggestion, @nikhilnanivadekar. It makes more sense now.
Let me know if I'm still missing anything. Would gladly amend.

**Comment by nikhilnanivadekar** [comment]
LGTM

**Pull request merged**